src/useForm.ts

```diff
@@ -165,14 +166,43 @@ export class FormStore {
     return cache;
   };
 
-  private getFieldsValue = (nameList?: NamePath[]) => {
+  private getFieldsValue = (
+    nameList?: NamePath[],
+    filterFunc?: (meta: Meta) => boolean,
+  ) => {
     this.warningUnhooked();
 
-    if (!nameList) {
+    if (!nameList && !filterFunc) {
       return this.store;
     }
+    if (!filterFunc) {
+      return cloneByNamePathList(this.store, nameList.map(getNamePath));
+    }
+
+    let fieldEntities = this.getFieldEntities(true);
+
+    if (nameList) {
+      const cache = this.getFieldsMap(true);
+
+      fieldEntities = nameList.map(name => {
+        const namePath = getNamePath(name);
+        return cache.get(namePath);
+      });
+    }
+
+    const filteredNameList = fieldEntities.reduce(
+      (filteredNameListSoFar, field: FieldEntity): NamePath[] => {
+        const namePath = field.getNamePath();
+        const meta = field.getMeta();
+        if (filterFunc(meta)) {
+          filteredNameListSoFar.push(namePath);
+        }
+        return filteredNameListSoFar;
```

Comment:
Seems here use same array instead of return new one in reduce. We can use just for each loop to handle this.